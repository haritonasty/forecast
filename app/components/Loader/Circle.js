import styled, { keyframes } from 'styled-components';
import { colors } from './colors';

const Circle = styled.div`
  width: 7px;
  height: 7px;
  border-radius: 50%;
  background-color: ${colors.blue};
`;

export const bouncingLoader = keyframes`
  to {
    transform: translate3d(0, -1rem, 0);
  }
`;

export default Circle;
