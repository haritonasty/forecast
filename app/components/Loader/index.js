/**
 *
 * Loader
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Circle, { bouncingLoader } from './Circle';
import { colors } from './colors';
import Wrapper from './Wrapper';

const StyledWrapper = styled(Wrapper)`
  & > ${Circle} {
    margin: 2px;
    animation: ${bouncingLoader} 500ms infinite alternate;
    
    &:nth-child(2) {
      background: ${colors.red};
      animation-delay: 125ms;
    }
    
    &:nth-child(3) {
      background: ${colors.yellow};
      animation-delay: 250ms;
    }
    
    &:nth-child(4) {
      background: ${colors.green};
      animation-delay: 375ms;
    }
  }
}
`;

function Loader({ className }) {
  return (
    <StyledWrapper className={className}>
      <Circle />
      <Circle />
      <Circle />
      <Circle />
    </StyledWrapper>
  );
}

Loader.propTypes = {
  className: PropTypes.string,
};

export default Loader;
