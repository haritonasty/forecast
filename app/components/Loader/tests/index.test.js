import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Loader from '../index';

describe('<Loader />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(<Loader />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
