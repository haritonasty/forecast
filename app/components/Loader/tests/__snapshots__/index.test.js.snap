// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should match the snapshot 1`] = `
.c2 {
  width: 7px;
  height: 7px;
  border-radius: 50%;
  background-color: #4285f4;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 60px;
  width: 100%;
}

.c0 > .c1 {
  margin: 2px;
  -webkit-animation: gsZCoC 500ms infinite alternate;
  animation: gsZCoC 500ms infinite alternate;
}

.c0 > .c1:nth-child(2) {
  background: #ea4335;
  -webkit-animation-delay: 125ms;
  animation-delay: 125ms;
}

.c0 > .c1:nth-child(3) {
  background: #fbbc05;
  -webkit-animation-delay: 250ms;
  animation-delay: 250ms;
}

.c0 > .c1:nth-child(4) {
  background: #34a853;
  -webkit-animation-delay: 375ms;
  animation-delay: 375ms;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  />
  <div
    className="c1 c2"
  />
  <div
    className="c1 c2"
  />
  <div
    className="c1 c2"
  />
</div>
`;
